tests/ResponseTest.php

```diff
@@ -95,6 +116,69 @@ public function basicResponsesDataProvider(): array
             ['invoice' => 23, 'status' => 'pending'],
           ],
 
+          'respondWithSuccess(), empty collection' => [
+            'respondWithSuccess',
+            [new Collection()],
+            Response::HTTP_OK,
+            ['success' => true],
+          ],
+
+          'respondWithSuccess(), Arrayable' => [
+            'respondWithSuccess',
+            [
+              new class implements Arrayable {
+                public function toArray()
+                {
+                  return ['id' => 1, 'name' => 'John'];
+                }
+              },
+            ],
+            Response::HTTP_OK,
+            ['id' => 1, 'name' => 'John']
+          ],
+
+          'respondWithSuccess(), empty Arrayable' => [
+            'respondWithSuccess',
+            [
+              new class implements Arrayable {
+                public function toArray()
+                {
+                  return [];
+                }
+              },
+            ],
+            Response::HTTP_OK,
+            ['success' => true]
+          ],
```

Comment:
Those didn't fail before my changes, however if you would break morphToArray on purpose for Arrayable only then that would not be caught by tests. That's why I added those.